### PR TITLE
Prepare docs for November 2022 wave of Providers

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/alibaba/CHANGELOG.rst
Comment (eladkal) [inline airflow/providers/alibaba/CHANGELOG.rst:36]: When bumping to 2.2 we did it as a breaking change.
Why bumping to 2.3 is a feature change?

File: airflow/providers/slack/CHANGELOG.rst
Comment (eladkal) [inline airflow/providers/slack/CHANGELOG.rst]: Should be  with 7.0.0 ( there is an entry in change log)

File: airflow/providers/snowflake/CHANGELOG.rst
Comment (eladkal) [inline airflow/providers/snowflake/CHANGELOG.rst]: Should be 4.0.0 ( there is 5.0 entry in log but i guess it meant to ve 4.0)

File: airflow/providers/tableau/provider.yaml
Comment (eladkal) [inline airflow/providers/tableau/provider.yaml]: No need for 3.1.0

File: airflow/providers/slack/CHANGELOG.rst
Comment (potiuk) [reply airflow/providers/slack/CHANGELOG.rst]: Oh thanks. I was going to review those today as well :)

File: airflow/providers/alibaba/CHANGELOG.rst
Comment (potiuk) [reply airflow/providers/alibaba/CHANGELOG.rst:36]: Very good question. I was waiting for it :)

I think it was a mistake to bump major version (But I can be convinced otherwise and change it).

I think this is a perfect representation of the view I have on distinguishing the breaking from non-breaking change and Hyrum's Law - https://www.hyrumslaw.com/. Every change is technically breaking someone's workflow. Breaking change is not a "0/1" decision but a continuum and it's mainly up to us to decide how to classify it, based on likelihood it will break many user's workflows.

The rationale i have - that from the point of view of provider, it's just a dependency change (which we generally consided non-breaking) and it does not break people's workflows in general at all.

* If you are on Airflow 2.2, you cannot install this release anyway. So the change is not breking - from the point of view of Airflow 2.2 user, there is no change. The user will have to continue using old provider.
  
* If you are on Airflow 2.3, you install it and it "just works". No breaking in your workflow whatsoever. 

There are two questions/potential problems/considerations  here:

1) You could accidentally bring your airflow up if you have Airflow 2.2 and force installation of 2.3+provider. I'd argue it's a problem of the user and the most we can do it is to warn user about it in the documentation. This is also true for many of our dependencies. There are a number of dependencies that - when upgraded independently - might bring airlfow version up (for example sqlalchemy bump to 1.4 will upgrade airflow to 2.3). Unfortunately `pip` works like that and we cannot do anything about it - everything that happens there happens long before any of our code is executed and we cannot even give a warning to the user when the user attempts to do so. Here potentially the "major" bump might give the user a pause, but I argue that this is a very weak signal - the major bump should signal incompatibilities in the provider itself, not potential problems with Airflow. Luckily this is not destructive nowadays. User will get information about migrations not applied and will be able to recover. 

2) I was thinking whether it shoudl be a "feature" or "patchlevel" bump. I chose "feature" pretty arbitrary here and without any good "reason". SemVer does not really address this case so we are free to choose whatever we feel is appropriate. And I think personally that signalling minor bump is a bit more prominent. We do not (I think) make it a breaking change (see above), but we also do not want to make it a "meh" change. We need to pick some scheme that will allow us to publish it as a new package in PyPI even without any changes. So I chose "feature". But it could be "patchlevel" as well. Both "patchlevel" and "feature" are not breaking (probably) user's workflow so choosing one over another is pretty arbitrary decision.

I wonder what others think in this context.

Comment (eladkal) [reply airflow/providers/alibaba/CHANGELOG.rst:36]: Thinking of it I tend to agree it shouldnt be a breaking change.
However specificly for this release if we make it feature we need to be extra carful reviewing the change log since we started introducing breaking changes to peoviders under the assumption all providers are going to have major release.

I aslo think if we change to feature telease from now on we should document it and maybe announce it as we are changing a previous convention.

Comment (potiuk) [reply airflow/providers/alibaba/CHANGELOG.rst:36]: I think we did not have a convention for "features". And we are not changing "regular" features. We are simply deciding how we treat bumping min version.

Comment (potiuk) [reply airflow/providers/alibaba/CHANGELOG.rst:36]: > However specificly for this release if we make it feature we need to be extra carful reviewing the change log since we started introducing breaking changes to peoviders under the assumption all providers are going to have major release.

Oh, absolutely - I just prepared it yesterday at 2 am to get it out but I will do  a round of reviews and rebases before I merge it (as usual :) . And great that others will also  help in identifying some of the mistakes :).

File: airflow/providers/tableau/provider.yaml
Comment (potiuk) [reply airflow/providers/tableau/provider.yaml]: Good point.

File: airflow/providers/snowflake/CHANGELOG.rst
Comment (potiuk) [reply airflow/providers/snowflake/CHANGELOG.rst]: Yep.

Comment (potiuk) [reply airflow/providers/snowflake/CHANGELOG.rst]: Yep.

File: airflow/providers/alibaba/CHANGELOG.rst
Comment (potiuk) [reply airflow/providers/alibaba/CHANGELOG.rst:36]: > I aslo think if we change to feature telease from now on we should document it and maybe announce it as we are changing a previous convention.

Ah I see what you mean (misunderstood it) - yeah. I am going to be rather "explicit" about how we treat such bumps (not features) it in devlist and will add some note to our README.

**Comment by potiuk** [comment]
Ok. Reviewed, updated, added also some more prominent "2.3+" message. I will add some comunication on devlist too in a moment.

**Comment by potiuk** [comment]
I also added clarification in our README about this new approach (and raise it in the devlist).

**Comment by potiuk** [comment]
I reviewed and merged all things I think should be included in this release - looking forward for an approval.

File: airflow/providers/airbyte/CHANGELOG.rst
Comment (jedcunningham) [inline airflow/providers/airbyte/CHANGELOG.rst:36]: Not requiring a major bump makes sense to me. It does, however, make me wonder if we should even do a release for the providers where the only change is this minimum?

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: Yes I think we should because otherwise we will have to selectively do it every time when we will track any change that is 2.3 incompatible. It makes it much cleaner and easier to manage and creates far less overhead when preparing the release wave. And It gives the user no more burden - most of the users are upgrading together with Airlfow - which means that they will automatically use the new providers.

It also provides a much better starting point in the future when we will have to (for example for security reasons) release a catastrofic fix in a proivider that should be installed for pre-2.2 version. We know exactly where to branch off, we still have patch-levels to use for that.

Most of our users do not follow and release providers when they are released (and we don't even encourage it). The big benefit of releasing provideres independently, is that the users CAN do that when they want. 

Also releasing it now gives a strong signal - there will be no more feature releases for all providers with pre-2.3 version. There will be at most bugfixes. If we don't upgrade now, there is still an option to release feature version for those providers that is not 2.2 compatible, which we should not do any more. And this is yet another signal to our users that they should upgrade to Airflow 2.3 asap. 

So yes - I think releasing all providers now makes perfect sense. Mostly that this is a clear communication "no more feature provider releases that will support Airflow 2.2"

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: But - just to comment on that as well @jedcunningham  - I thought also why I am doing it this way.

There is another good reason why we should release them all at least this time (but we can think about changing it in the future). 

Releasing "all" providers in such cases is simply much simpler and less risky and error-prone for me as release manager. Currently you would have to manually go through pretty much all those packages and find out where minor upgrade is beacuase we have "bump" and where there are "real features".

It's just way more work for me and the kind of work is far more prone to manual mistakes (unless we automate this scenario somehow, which makes it far more stresful - because being perfectiionist, I will likely have to check it 4 or five times if I made any manual mistake or not.  And knowing myself some mistake will slip through anyway. I am simply human.

So If we are to do that, I would likely have to first implement some automation to do it, otherwise I will find myself in a situation that I do not really like, where I have a working, automated process which has a number of safeguards in place that will prevent from making stupid mistakes. And likely it's better to do it for the next wave.

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: Just to give a very good example of propblems if we do not release "all" providers. 
Real example from just now. Before rebase I am doing now) someone merged:

* ``Replace urlparse with urlsplit (#27389)``

This change is a non-breaking, non feature, "misc" kind of change, that we - regardless - shoud put in the release docs and release, just in case this caused some problems or changes we are not aware of (it's not a bugfix, not a feature, but it changes code so potentially we should be able to explain that it was the reason when someone has some problems.

But this change was touching 7 providers. This means that it *could* have changed whether we release a provider or no (previously it was min-version only, now it became min-version + urlparse).

Should we release it now ? I think yes - the longer we wait with realeasing any code change the better. If we hold with relasing of provider that already has some "code" change" - then if we release it in 3 months we might have detect some problems that we could hve detected 3 months earlier. 

By having "release all changed in any way except documentation" providers I free myself from having to review each of those manually and decide - I release this/I do not release that. Currently we automatically skip from release only providers that have "doc-only changes" - for those we only regenerate the documentation. And this is all automated. If we were to decide "and we do not realease also that change" - we would have to implement automation for exclusion of those changes.

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: Simmply the more manual decision points, the less automation - the more "mistakes" we will do when releasing NN+ providers at the same time.

Comment (jedcunningham) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: Thanks @potiuk, yep that all makes sense. I was simply looking at if from the users side initially, but I definitely see value in keeping it simple for us.

File: airflow/providers/airbyte/CHANGELOG.rst
Comment (jedcunningham) [inline airflow/providers/airbyte/CHANGELOG.rst]:
```suggestion
This release of provider is only available for Airflow 2.3+ as explained in the
`Apache Airflow providers support policy <https://github.com/apache/airflow/blob/main/README.md#support-for-providers>`_.
```

Probably better if we use a link here?

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst]: Right. I am hesitant in putting any markup in CHANGELOG - because it becomes part of the README.rst and eventually becomes part of the package in PyPI - but links are save and they should both render correctly and be accepted by PyPI (we will see if they are passing our CI - we have `twine` check as part of our CI and they will fail if the README's are not complying with `PyPI` expectations.

Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst]: Updated - also for past changelog entries.

File: airflow/providers/airbyte/CHANGELOG.rst
Comment (potiuk) [reply airflow/providers/airbyte/CHANGELOG.rst:36]: > Thanks @potiuk, yep that all makes sense. I was simply looking at if from the users side initially, but I definitely see value in keeping it simple for us.

I knew you would :)

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
Should be fine. Merging :)

**Pull request merged**
